arwen/elrondapitest/bigFloats_test.go

```diff
@@ -0,0 +1,690 @@
+				resultMul := new(big.Float).Mul(bigFloatValue, bigFloatValue)
+				bigFloatValue.Set(resultMul)
+			}
+			floatBuffer, _ := bigFloatValue.GobEncode()
+			fmt.Println(floatBuffer)
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_FailExecution_Mul(t *testing.T) {
+	numberOfReps := 30
+	repsArgument := []byte{0, 0, 0, byte(numberOfReps)}
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatMulTest").
+			WithArguments(repsArgument,
+				floatArgument1).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			verify.
+				ReturnCode(10).
+				ReturnData()
+		})
+}
+
+func TestBigFloats_Div(t *testing.T) {
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatDivTest").
+			WithArguments(repsArgument,
+				floatArgument1,
+				floatArgument2).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			numerator := new(big.Float)
+			_ = numerator.GobDecode(floatArgument1)
+			denominator := new(big.Float)
+			_ = denominator.GobDecode(floatArgument2)
+			for i := 0; i < numberOfReps; i++ {
+				resultMul := new(big.Float).Quo(numerator, denominator)
+				numerator.Set(resultMul)
+			}
+			floatBuffer, _ := numerator.GobEncode()
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_Truncate(t *testing.T) {
+	test.BuildInstanceCallTest(t).
+		WithContracts(
+			test.CreateInstanceContract(test.ParentAddress).
+				WithCode(test.GetTestSCCode("big-floats", "../../"))).
+		WithInput(test.CreateTestContractCallInputBuilder().
+			WithGasProvided(100000).
+			WithFunction("BigFloatTruncateTest").
+			WithArguments(repsArgument,
+				floatArgument1,
+				floatArgument2).
+			Build()).
+		AndAssertResults(func(host arwen.VMHost, stubBlockchainHook *contextmock.BlockchainHookStub, verify *test.VMOutputVerifier) {
+			value1 := big.NewFloat(0)
+			_ = value1.GobDecode(floatArgument1)
+			value2 := big.NewFloat(0)
+			_ = value2.GobDecode(floatArgument2)
+			for i := 0; i < numberOfReps; i++ {
+				rDiv := big.NewInt(0)
+				value1.Int(rDiv)
+				result := big.NewFloat(0).Sub(value1, value2)
+				value1.Set(result)
+			}
+			floatBuffer, _ := value1.GobEncode()
+			verify.
+				Ok().
+				ReturnData(
+					floatBuffer)
+		})
+}
+
+func TestBigFloats_Abs(t *testing.T) {
+	bigFloatArguments := make([][]byte, numberOfReps+1)
+	for i := range bigFloatArguments {
```

Comment:
The contract `big-floats.c` only returns the absolute value of the last float. Consider either:
* change the contract to return the absolute values of all the floats and write assertions for all of them
* leave the contract as-is, but specify `numberOfReps = 1` for this test, instead of creating a long slice of arguments that the contract must process, but for which there's no assertion